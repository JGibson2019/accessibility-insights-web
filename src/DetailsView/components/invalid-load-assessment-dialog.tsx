// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { Dialog, DialogFooter, DialogType, PrimaryButton } from 'office-ui-fabric-react';
import * as styles from 'DetailsView/components/invalid-load-assessment-dialog.scss';
import * as React from 'react';

export interface InvalidLoadAssessmentDialogProps {
    isOpen: boolean;
    onClose: () => void;
}

export const InvalidLoadAssessmentDialog = NamedFC<InvalidLoadAssessmentDialogProps>(
    'InvalidLoadAssessmentDialog',
    props => {
        const onDismiss = (): void => {
            props.onClose();
        };
        if (!props.isOpen) {
            return null;
        }

        return (
            <Dialog
                hidden={false}
                //JG note: clean this up and sub in onClose if we don't have something to add on line 15
                onDismiss={onDismiss}
                dialogContentProps={{
                    type: DialogType.normal,
                    title: 'Oops! Accessibility Insights canâ€™t open this file.',
                    subText: 'Please select a valid *.a11yAssessment file',
                }}
                modalProps={{
                    isBlocking: false,
                    containerClassName: styles.invalidLoadAssessmentDialog,
                    onDismissed: props.onClose,
                }}
            >
                <DialogFooter>
                    <PrimaryButton onClick={props.onClose}>OK</PrimaryButton>
                </DialogFooter>
            </Dialog>
        );
    },
);
