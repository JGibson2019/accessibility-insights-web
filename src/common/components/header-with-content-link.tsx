// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as styles from 'common/components/header-with-content-common.scss';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import { ContentLink, ContentLinkDeps } from 'views/content/content-link';
import { ContentReference } from 'views/content/content-page';

export interface HeaderWithContentLinkProps {
    deps: ContentLinkDeps;
    headerClass?: string;
    headingTitleClassName?: string;
    headingTitle: string;
    secondaryText?: string;
    guidance?: ContentReference;
}

export const HeaderWithContentLink = NamedFC<HeaderWithContentLinkProps>(
    'HeaderWithContent Link',
    props => {
        return (
            <div className={`${styles.headerWithContentLink} ${props.headerClass}`}>
                <h1>
                    <span
                        className={props.headingTitleClassName}
                    >{`${props.headingTitle} ${props.secondaryText}`}</span>
                </h1>
                <ContentLink deps={props.deps} reference={props.guidance} iconName="info" />
            </div>
        );
    },
);
