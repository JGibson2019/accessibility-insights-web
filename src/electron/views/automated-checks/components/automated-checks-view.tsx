// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

export type AutomatedChecksViewDeps = {};

export type AutomatedChecksViewProps = {
    deps: AutomatedChecksViewDeps;
};

export class AutomatedChecksView extends React.Component<AutomatedChecksViewProps> {
    public render(): JSX.Element {
        return <>automated checks</>;
    }
}
