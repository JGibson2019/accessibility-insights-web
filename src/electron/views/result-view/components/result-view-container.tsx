import { WindowStateStore } from 'electron/flux/store/window-state-store';
import { ClientStoresHub } from 'common/stores/client-stores-hub';

// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export type ResultViewContainerDeps = {
    storeHub: ClientStoresHub<WindowStateStore>;
};

export type ResultViewContainerProps = {
    deps: ResultViewContainerDeps;
};
