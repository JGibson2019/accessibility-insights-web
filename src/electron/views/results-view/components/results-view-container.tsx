// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import { ClientStoresHub } from 'common/stores/client-stores-hub';
import { WindowStateStore } from 'electron/flux/store/window-state-store';

export type ResultsViewContainerDeps = {};

export type ResultsViewContainerProps = {
    deps: ResultsViewContainerDeps;
};

export type ResultViewContainerState = {};

export class ResultsViewContainer extends React.Component<ResultsViewContainerProps, ResultViewContainerState> {
    constructor(props: ResultsViewContainerProps) {
        super(props);
    }

    public render(): JSX.Element {
        return null;
    }
}
