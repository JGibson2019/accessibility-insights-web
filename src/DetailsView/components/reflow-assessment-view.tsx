// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDefaultMessageGenerator } from 'assessments/assessment-default-message-generator';
import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { AssessmentViewUpdateHandler } from 'DetailsView/components/assessment-view-update-handler';
import * as styles from 'DetailsView/components/assessment-view.scss';
import * as React from 'react';
import { ContentLink, ContentLinkDeps } from 'views/content/content-link';
import { ContentPageComponent } from 'views/content/content-page';

import { AssessmentTestResult } from '../../common/assessment/assessment-test-result';
import { CollapsibleComponent } from '../../common/components/collapsible-component';
import { reactExtensionPoint } from '../../common/extensibility/react-extension-point';
import { Tab } from '../../common/itab';
import {
    AssessmentData,
    AssessmentNavState,
    PersistedTabInfo,
} from '../../common/types/store-data/assessment-result-data';
import { FeatureFlagStoreData } from '../../common/types/store-data/feature-flag-store-data';
import { PathSnippetStoreData } from '../../common/types/store-data/path-snippet-store-data';
import { DetailsViewActionMessageCreator } from '../actions/details-view-action-message-creator';
import { DetailsViewExtensionPoint } from '../extensions/details-view-extension-point';
import { AssessmentInstanceTableHandler } from '../handlers/assessment-instance-table-handler';
import { GettingStartedView, GettingStartedViewProps } from './getting-started-view';
import { TargetChangeDialog, TargetChangeDialogDeps } from './target-change-dialog';
import { TestStepView, TestStepViewDeps } from './test-step-view';
import { TestStepNavDeps, TestStepsNav } from './test-steps-nav';

export type WithAssessmentTestResult = { assessmentTestResult: AssessmentTestResult };
export const AssessmentViewMainContentExtensionPoint = reactExtensionPoint<
    WithAssessmentTestResult
>('AssessmentViewMainContentExtensionPoint');

export type AssessmentViewDeps = ContentLinkDeps &
    TestStepViewDeps &
    TestStepNavDeps &
    TargetChangeDialogDeps & {
        detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
        assessmentsProvider: AssessmentsProvider;
        assessmentViewUpdateHandler: AssessmentViewUpdateHandler;
        detailsViewExtensionPoint: DetailsViewExtensionPoint;
    };

export interface AssessmentViewProps {
    deps: AssessmentViewDeps;
    isScanning: boolean;
    isEnabled: boolean;
    assessmentNavState: AssessmentNavState;
    assessmentInstanceTableHandler: AssessmentInstanceTableHandler;
    assessmentData: AssessmentData;
    currentTarget: Tab;
    prevTarget: PersistedTabInfo;
    assessmentDefaultMessageGenerator: AssessmentDefaultMessageGenerator;
    assessmentTestResult: AssessmentTestResult;
    featureFlagStoreData: FeatureFlagStoreData;
    pathSnippetStoreData: PathSnippetStoreData;
}

export class ReflowAssessmentView extends React.Component<AssessmentViewProps> {
    public static readonly requirementsTitle: string = 'Requirements';

    private deps: AssessmentViewDeps;

    constructor(props: AssessmentViewProps) {
        super(props);
        this.deps = props.deps;
        this.state = {
            isStartOverDialogOpen: false,
        };
    }

    public render(): JSX.Element {
        const { assessmentTestResult } = this.props;
        const { extensions } = assessmentTestResult.definition;
        const extPointProps = { extensions, assessmentTestResult };
        return (
            <div className={styles.assessmentContent}>
                {this.renderTargetChangeDialog()}
                {this.renderTitle(
                    assessmentTestResult.definition.title,
                    assessmentTestResult.definition.guidance,
                )}
                {this.renderGettingStarted(assessmentTestResult.definition.gettingStarted)}
                <AssessmentViewMainContentExtensionPoint.component {...extPointProps}>
                    {this.renderRequirements()}
                    {this.renderMainContent(assessmentTestResult)}
                </AssessmentViewMainContentExtensionPoint.component>
            </div>
        );
    }

    public componentDidMount(): void {
        this.deps.assessmentViewUpdateHandler.onMount(this.props);
    }

    public componentDidUpdate(prevProps: AssessmentViewProps): void {
        this.deps.assessmentViewUpdateHandler.update(prevProps, this.props);

        const { assessmentTestResult } = this.props;
        this.deps.detailsViewExtensionPoint
            .apply(assessmentTestResult.definition.extensions)
            .onAssessmentViewUpdate(prevProps, this.props);
    }

    public componentWillUnmount(): void {
        this.deps.assessmentViewUpdateHandler.onUnmount(this.props);
    }

    private renderTargetChangeDialog(): JSX.Element {
        return (
            <TargetChangeDialog
                deps={this.props.deps}
                prevTab={this.props.prevTarget}
                newTab={this.props.currentTarget}
            />
        );
    }

    private renderTitle(title: string, content?: ContentPageComponent): JSX.Element {
        return (
            <div className={styles.assessmentTitle}>
                <h1 className={styles.assessmentHeader}>
                    {title} <ContentLink deps={this.deps} reference={content} iconName="info" />
                </h1>
            </div>
        );
    }

    private renderGettingStarted(gettingStarted: JSX.Element): JSX.Element {
        return (
            <CollapsibleComponent
                header={<h2>Getting Started</h2>}
                content={gettingStarted}
                containerClassName={styles.assessmentGettingStartedContainer}
            />
        );
    }

    private renderRequirements(): JSX.Element {
        return (
            <div className={styles.assessmentRequirementsTitle}>
                <h2 className={styles.assessmentRequirements}>
                    {ReflowAssessmentView.requirementsTitle}
                </h2>
            </div>
        );
    }

    private renderMainContent(assessmentTestResult: AssessmentTestResult): JSX.Element {
        if (this.props.assessmentNavState.selectedTestSubview === 'getting-started') {
            const props: GettingStartedViewProps = {
                gettingStartedContent: (
                    <div>
                        Could pass the getting started content for the requirement as part of props
                    </div>
                ),
            };
            return (
                <div className={styles.detailsViewAssessmentContent}>
                    <GettingStartedView {...props} />
                </div>
            );
        } else {
            const selectedRequirement = assessmentTestResult.getRequirementResult(
                this.props.assessmentNavState.selectedTestSubview,
            );
            const isStepScanned = selectedRequirement.data.isStepScanned;
            return (
                <div className={styles.detailsViewAssessmentContent}>
                    <div className={styles.testStepsNavContainer}>
                        <TestStepsNav
                            deps={this.props.deps}
                            ariaLabel={ReflowAssessmentView.requirementsTitle}
                            selectedTest={assessmentTestResult.visualizationType}
                            selectedTestStep={selectedRequirement.definition.key}
                            stepStatus={this.props.assessmentData.testStepStatus}
                        />
                    </div>
                    <div className={styles.testStepViewContainer}>
                        <TestStepView
                            deps={this.deps}
                            isScanning={this.props.isScanning}
                            testStep={selectedRequirement.definition}
                            renderStaticContent={selectedRequirement.definition.renderStaticContent}
                            instancesMap={this.props.assessmentData.generatedAssessmentInstancesMap}
                            assessmentNavState={this.props.assessmentNavState}
                            assessmentInstanceTableHandler={
                                this.props.assessmentInstanceTableHandler
                            }
                            manualTestStepResultMap={
                                this.props.assessmentData.manualTestStepResultMap
                            }
                            assessmentsProvider={this.props.deps.assessmentsProvider}
                            isStepEnabled={this.props.isEnabled}
                            isStepScanned={isStepScanned}
                            assessmentDefaultMessageGenerator={
                                this.props.assessmentDefaultMessageGenerator
                            }
                            featureFlagStoreData={this.props.featureFlagStoreData}
                            pathSnippetStoreData={this.props.pathSnippetStoreData}
                        />
                    </div>
                </div>
            );
        }
    }
}
