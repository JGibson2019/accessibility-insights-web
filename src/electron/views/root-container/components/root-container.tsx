// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import React from 'react';

import { TelemetryPermissionDialogDeps } from 'common/components/telemetry-permission-dialog';
import { ClientStoresHub } from 'common/stores/client-stores-hub';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import { DeviceStoreData } from 'electron/flux/types/device-store-data';
import { WindowStateStoreData } from 'electron/flux/types/window-state-store-data';
import { DeviceConnectBodyDeps } from 'electron/views/device-connect-view/components/device-connect-body';
import { DeviceConnectViewContainer } from 'electron/views/device-connect-view/components/device-connect-view-container';
import {
    ResultsViewContainer,
    ResultsViewContainerDeps,
    ResultsViewContainerProps,
} from 'electron/views/results-view/components/results-view-container';

// root container deps should only depend on ResultsViewContainerDeps and DeviceConnectBodyDeps. This will be addresed in a later PR
export type RootContainerDeps = ResultsViewContainerDeps &
    TelemetryPermissionDialogDeps &
    DeviceConnectBodyDeps & {
        storeHub: ClientStoresHub<RootContainerState>;
    };

export type RootContainerProps = ResultsViewContainerProps & {
    deps: RootContainerDeps;
};

export type RootContainerState = {
    windowStateStoreData: WindowStateStoreData;
    userConfigurationStoreData: UserConfigurationStoreData;
    deviceStoreData: DeviceStoreData;
};

export class RootContainer extends React.Component<RootContainerProps, RootContainerState> {
    constructor(props: RootContainerProps) {
        super(props);

        this.state = props.deps.storeHub.getAllStoreData();
    }

    public render(): JSX.Element {
        if (this.state.windowStateStoreData.routeId === 'resultsView') {
            return <ResultsViewContainer {...this.props}></ResultsViewContainer>;
        }
        return <DeviceConnectViewContainer {...this.props}></DeviceConnectViewContainer>;
    }

    public componentDidMount(): void {
        this.props.deps.storeHub.addChangedListenerToAllStores(this.onStoresChange);
    }

    private onStoresChange = () => {
        this.setState(() => this.props.deps.storeHub.getAllStoreData());
    };
}
