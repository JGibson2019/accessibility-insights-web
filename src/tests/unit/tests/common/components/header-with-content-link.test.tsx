// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { HeaderWithContentLink } from 'common/components/header-with-content-link';
import { shallow } from 'enzyme';
import * as React from 'react';

import { ContentPage } from 'views/content/content-page';
import { ContentActionMessageCreator } from '../../../../../common/message-creators/content-action-message-creator';

describe('HeaderWithContentLink', () => {
    const contentPath = 'for/testing';
    const content = {
        for: {
            testing: ContentPage.create(() => 'CONTENT FOR TESTING' as any),
        },
    };

    const contentProvider = ContentPage.provider(content);

    const openContentPage = jest.fn();

    const contentActionMessageCreator = {
        openContentPage,
    } as Partial<ContentActionMessageCreator> as ContentActionMessageCreator;

    const deps = {
        contentProvider,
        contentActionMessageCreator,
    };

    const headerTitle = 'header';

    const secondaryText = 'secondary';

    it('render null when reference is not defined', () => {
        const result = shallow(
            <HeaderWithContentLink deps={deps} guidance={null} headingTitle={headerTitle} />,
        );
        expect(result.debug()).toMatchSnapshot();
    });

    it('renders with both text and icon', () => {
        const result = shallow(
            <HeaderWithContentLink
                deps={deps}
                guidance={contentPath}
                headingTitle={headerTitle}
                iconName="test icon"
            />,
        );
        expect(result.debug()).toMatchSnapshot();
    });

    it('renders with heading, secondary text, and icon', () => {
        const result = shallow(
            <HeaderWithContentLink
                deps={deps}
                guidance={contentPath}
                headingTitle={headerTitle}
                secondaryText={secondaryText}
                iconName="test icon"
            />,
        );
        expect(result.debug()).toMatchSnapshot();
    });
});
