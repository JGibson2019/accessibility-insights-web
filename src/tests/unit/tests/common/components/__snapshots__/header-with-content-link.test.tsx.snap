// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWithContentLink render null when reference is not defined 1`] = `
"<div className=\\"headerWithContentLink undefined\\">
  <h1>
    <span className={[undefined]}>
      header undefined
    </span>
  </h1>
  <ContentLink deps={{...}} reference={{...}} iconName=\\"info\\" />
</div>"
`;

exports[`HeaderWithContentLink renders with both text and icon 1`] = `
"<div className=\\"headerWithContentLink undefined\\">
  <h1>
    <span className={[undefined]}>
      header undefined
    </span>
  </h1>
  <ContentLink deps={{...}} reference=\\"for/testing\\" iconName=\\"info\\" />
</div>"
`;

exports[`HeaderWithContentLink renders with heading, secondary text, and icon 1`] = `
"<div className=\\"headerWithContentLink undefined\\">
  <h1>
    <span className={[undefined]}>
      header secondary
    </span>
  </h1>
  <ContentLink deps={{...}} reference=\\"for/testing\\" iconName=\\"info\\" />
</div>"
`;
